pennylane/transforms/diagonalize_measurements.py

```diff
@@ -60,6 +60,14 @@ def diagonalize_measurements(tape, supported_base_obs=_default_supported_obs, to
         circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>` transform
         can be applied.
 
+    .. note::
+        This transform will diagonalize what it can, i.e., ``qml.X``, ``qml.Y``, ``qml.Z``,
+        ``qml.Hadamard``, ``qml.Identity``, or a linear combination of them. Any unrecognized
+        observable will not raise an error, deferring to the device's validation for supported
+        measurements later on. Lastly, if ``diagonalize_measurements`` produces additional gates
+        that the device does not support, the :func:`~pennylane.devices.preprocess.decompose``
+        transform should be applied to ensure that the additional gates are decomposed to those
```

Comment:
```suggestion
        measurements later on. 
        
        Lastly, if you are including ``diagonalize_measurements`` as part of the transform program
        constructed in a custom implementation of ``Device.preprocess_transforms`` for your
        plugin device, note that the transform may produce additional gates that the device does 
        not support, so the :func:`~pennylane.devices.preprocess.decompose`` transform should 
        be applied after to ensure that the additional gates are decomposed to those
```
@isaacdevlugt  Is this better?